src/chart/graph/categoryVisual.js

```diff
@@ -88,6 +100,27 @@ export default function (ecModel) {
                         );
                     }
 
+                    if (!data.getItemVisual(idx, 'symbolKeepAspect', true)) {
```

Comment:
Maybe a for-loop here?

Reply:
Ummm, loop for what…？

Reply:
`['symbol', 'symbolSize', ''symbolKeepAspect', ...]`, all similar logic.

Reply:
Already done :-)